File: app/src/main/java/com/blues/module_gank/old/StudyPageFragment.java
package com.blues.module_gank.old;

import android.content.Context;
import android.os.Bundle;
import android.view.Gravity;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.blues.R;
import com.blues.framework.base.BaseFragment;
import com.blues.framework.base.BaseViewPagerAdapter;
import com.blues.module_gank.old.model_android.view.AndroidFragment;
import com.blues.module_gank.old.model_article.ArticleFragment;
import com.blues.module_gank.old.model_ios.view._iOSFragment;
import com.blues.module_gank.old.model_video.VideoFragment;
import com.blues.module_gank.old.model_web.view.WebFragment;
import com.google.android.material.tabs.TabLayout;

import java.util.Arrays;

import androidx.annotation.NonNull;
import androidx.viewpager.widget.ViewPager;


/**
 * User : Blues
 * Date : 2019/4/11
 * Time : 15:32
 */

public class StudyPageFragment extends BaseFragment {

    private String[] tabs = {"Video", "Android", "iOS", "Web", "Blues"};

    @Override
    public int setLayoutResourceId() {
        return R.layout.fragment_studypage;
    }

    @Override
    public void initLayout(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {

        TabLayout mTab = rootView.findViewById(R.id.tl_fragment_bottom);
        ViewPager mViewPager = rootView.findViewById(R.id.vp_show);

        BaseViewPagerAdapter mFragmentAdapter = new BaseViewPagerAdapter(getChildFragmentManager(), Arrays.asList(tabs));
        mFragmentAdapter.addFragment(new VideoFragment());
        mFragmentAdapter.addFragment(new AndroidFragment());
        mFragmentAdapter.addFragment(new _iOSFragment());
        mFragmentAdapter.addFragment(new WebFragment());
        mFragmentAdapter.addFragment(new ArticleFragment());

        mViewPager.setAdapter(mFragmentAdapter);
        mViewPager.setOffscreenPageLimit(1);

        mViewPager.addOnPageChangeListener(new TabLayout.TabLayoutOnPageChangeListener(mTab));

        mTab.addOnTabSelectedListener(new TabLayout.ViewPagerOnTabSelectedListener(mViewPager) {
            @Override
            public void onTabSelected(@NonNull TabLayout.Tab tab) {
                mViewPager.setCurrentItem(tab.getPosition());

                View view = tab.getCustomView();
                if (view instanceof TextView) {
                    ((TextView) view).setTextColor(view.getResources().getColor(R.color.color_light_blue));
                }
            }

            @Override
            public void onTabUnselected(TabLayout.Tab tab) {
                View view = tab.getCustomView();
                if (view instanceof TextView) {
                    ((TextView) view).setTextColor(view.getResources().getColor(R.color.color_black));
                }
            }

            @Override
            public void onTabReselected(TabLayout.Tab tab) {

            }
        });

        mTab.setSelectedTabIndicatorColor(mTab.getResources().getColor(R.color.color_light_blue));
        //因为fragment不销毁，所以添加新tabs前最好把旧的都移除掉先，否则就会出现重复的
        mTab.removeAllTabs();

        for (int i = 0; i < tabs.length; i++) {
            mTab.addTab(mTab.newTab().setCustomView(getTabView(getmContext(), tabs[i])), i == 0);
        }
    }

    @Override
    public void lazyFetchData() {

    }

    @Override
    public void setListener() {

    }

    private View getTabView(Context context, String text) {
        TextView textView = new TextView(context);
        textView.setGravity(Gravity.CENTER);
        textView.setText(text);
        return textView;
    }
}

File: app/src/main/java/com/blues/gankio/v1/StudyPageFragment.kt
package com.blues.gankio.v1

import android.content.Context
import com.blues.R
import android.view.LayoutInflater
import android.view.ViewGroup
import android.os.Bundle
import com.google.android.material.tabs.TabLayout
import androidx.viewpager.widget.ViewPager
import com.blues.framework.base.BaseViewPagerAdapter
import com.blues.gankio.v1.view.VideoFragment
import com.blues.gankio.v1.view.AndroidFragment
import com.blues.gankio.v1.view.IOSFragment
import com.blues.gankio.v1.view.WebFragment
import com.blues.gankio.v1.view.ArticleFragment
import com.google.android.material.tabs.TabLayout.TabLayoutOnPageChangeListener
import android.widget.TextView
import android.view.Gravity
import android.view.View
import com.blues.framework.base.BaseKoinFragment
import java.util.*

/**
 * User : Blues
 * Date : 2019/4/11
 * Time : 15:32
 */
class StudyPageFragment : BaseKoinFragment() {

    private val tabs = arrayOf("Video", "Android", "iOS", "Web", "Blues")
    private val fragments = arrayOf(VideoFragment(), AndroidFragment(), IOSFragment(), WebFragment(), ArticleFragment())

    private fun getTabView(context: Context, text: String): View {
        val textView = TextView(context)
        textView.gravity = Gravity.CENTER
        textView.text = text
        return textView
    }

    override fun getLayoutId(): Int = R.layout.fragment_studypage

    override fun initData(inflater: LayoutInflater, container: ViewGroup?, saveInstanced: Bundle?) {

        val mTab: TabLayout = rootView.findViewById(R.id.tl_fragment_bottom)
        val mViewPager: ViewPager = rootView.findViewById(R.id.vp_show)
        val mFragmentAdapter = BaseViewPagerAdapter(childFragmentManager, tabs.asList())
        fragments.map { mFragmentAdapter.addFragment(it) }
        mViewPager.apply {
            adapter = mFragmentAdapter
            offscreenPageLimit = 1
            addOnPageChangeListener(TabLayoutOnPageChangeListener(mTab))
        }

        mTab.apply {
            addOnTabSelectedListener(object : TabLayout.ViewPagerOnTabSelectedListener(mViewPager) {
                override fun onTabSelected(tab: TabLayout.Tab) {
                    mViewPager.currentItem = tab.position
                    val view = tab.customView
                    if (view is TextView) {
                        view.setTextColor(view.getResources().getColor(R.color.color_light_blue))
                    }
                }

                override fun onTabUnselected(tab: TabLayout.Tab) {
                    val view = tab.customView
                    if (view is TextView) {
                        view.setTextColor(view.getResources().getColor(R.color.color_black))
                    }
                }

                override fun onTabReselected(tab: TabLayout.Tab) {}
            })
            setSelectedTabIndicatorColor(mTab.resources.getColor(R.color.color_light_blue)) //因为fragment不销毁，所以添加新tabs前最好把旧的都移除掉先，否则就会出现重复的
            removeAllTabs()
        }
        for (i in tabs.indices) {
            mTab.addTab(mTab.newTab().setCustomView(getTabView(requireContext(), tabs[i])), i == 0)
        }
    }
}